atc/gc/build_log_retention_calculator.go

```diff
@@ -5,43 +5,69 @@ import (
 )
 
 type BuildLogRetentionCalculator interface {
-	BuildLogsToRetain(db.Job) int
+	BuildLogsToRetain(db.Job) (int, int)
 }
 
 type buildLogRetentionCalculator struct {
-	defaultBuildLogsToRetain uint64
-	maxBuildLogsToRetain     uint64
+	defaultBuildLogsToRetain     uint64
+	maxBuildLogsToRetain         uint64
+	defaultDaysToRetainBuildLogs uint64
+	maxDaysToRetainBuildLogs     uint64
 }
 
 func NewBuildLogRetentionCalculator(
 	defaultBuildLogsToRetain uint64,
 	maxBuildLogsToRetain uint64,
+	defaultDaysToRetainBuildLogs uint64,
+	maxDaysToRetainBuildLogs uint64,
 ) BuildLogRetentionCalculator {
 	return &buildLogRetentionCalculator{
-		defaultBuildLogsToRetain: defaultBuildLogsToRetain,
-		maxBuildLogsToRetain:     maxBuildLogsToRetain,
+		defaultBuildLogsToRetain:     defaultBuildLogsToRetain,
+		maxBuildLogsToRetain:         maxBuildLogsToRetain,
+		defaultDaysToRetainBuildLogs: defaultDaysToRetainBuildLogs,
+		maxDaysToRetainBuildLogs:     maxDaysToRetainBuildLogs,
 	}
 }
 
-func (blrc *buildLogRetentionCalculator) BuildLogsToRetain(job db.Job) int {
+func (blrc *buildLogRetentionCalculator) BuildLogsToRetain(job db.Job) (int, int) {
```

Comment:
Maybe this could return `atc.BuildLogRetention`? It's not clear which `int` is which, and we already have the handy struct for it.